bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java

```diff
@@ -183,11 +184,42 @@ private String getFirstLocalIPv4Address() {
         return broadcastAddresses;
     }
 
+    @Override
+    public String getPrimaryBroadcastAddress() {
+        String primaryIp = getPrimaryIpv4HostAddress();
+        String broadcastAddress = null;
+        if (primaryIp != null) {
+            try {
+                byte[] addr = InetAddress.getByName(primaryIp).getAddress();
+                Short prefix = getAllInterfaceAddresses().stream()
+                        .filter(a -> a.getAddress().getHostAddress().equals(primaryIp)).map(a -> a.getPrefix())
+                        .findFirst().get().shortValue();
+                byte[] netmask = InetAddress.getByName(networkPrefixLengthToNetmask(prefix)).getAddress();
+                byte[] broadcast = new byte[] { (byte) (~netmask[0] | addr[0]), (byte) (~netmask[1] | addr[1]),
+                        (byte) (~netmask[2] | addr[2]), (byte) (~netmask[3] | addr[3]) };
+                broadcastAddress = InetAddress.getByAddress(broadcast).getHostAddress();
+            } catch (UnknownHostException ex) {
+                LOGGER.error("Invalid IP address parameter: {}", ex.getMessage(), ex);
+            }
+        }
+        if (broadcastAddress == null) {
+            // an error has occurred, using broadcast address of first interface
+            broadcastAddress = getFirstIpv4BroadcastAddress();
+            LOGGER.warn(
+                    "Could not find broadcast address of configured IP, using broadcast address {} of first interface instead",
+                    broadcastAddress);
+        }
+        return broadcastAddress;
+    }
+
     /**
+     * Deprecated: Please use getPrimaryBroadcastAddress()
+     *
      * Get the first candidate for a broadcast address
      *
-     * @return broadcast address, null of no broadcast address is found
+     * @return broadcast address, null if no broadcast address is found
      */
+    @Deprecated
```

Comment:
Can you please add a comment that one should now use `getPrimaryBroadcastAddress` instead?

Reply:
The comment is in the java doc a few lines above.

Reply:
Sorry, my fault...